<?xml version="1.0"?>
<robot name="two_dof_robot" xmlns:xacro="http://ros.org/wiki/xacro">

  <!-- Base Link -->
  <link name="base"/>

  <!-- Joint 1 -->
  <joint name="joint1" type="revolute">
    <parent link="base"/>
    <child link="link1"/>
    <origin xyz="0 0 0.5" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.57" upper="1.57" effort="1.0" velocity="1.0"/>
  </joint>

  <!-- Link 1 -->
  <link name="link1">
    <visual>
      <origin xyz="0 0 0.0" rpy="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 1"/>
      </geometry>
    </visual>
  </link>

  <!-- Joint 2 -->
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0.05 0.525" rpy="0 1.5708 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="1.0" velocity="1.0"/>
  </joint>

  <!-- Link 2 -->
  <link name="link2">
    <visual>
      <origin xyz="0.05 0 0.5" rpy="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 1.05"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>

  <!-- Joint to End Effector (fixed) -->
  <joint name="joint_ee" type="fixed">
    <parent link="link2"/>
    <child link="end_effector"/>
    <origin xyz="0 0 0.525" rpy="0 0 0"/> <!-- link2 yüksekliği/2 -->
  </joint>

  <!-- End Effector Link -->
  <link name="end_effector">
    <visual>
      <origin xyz="0.05 0.0 0.5" rpy="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>

</robot>
